kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala

```diff
@@ -17,23 +17,40 @@
 
 package org.apache.kyuubi.operation
 
+import java.io.IOException
 import java.nio.ByteBuffer
 import java.util.{ArrayList => JArrayList, Locale}
 import java.util.concurrent.TimeUnit
 
 import scala.collection.JavaConverters._
 
+import com.codahale.metrics.MetricRegistry
 import org.apache.hive.service.rpc.thrift._
 
 import org.apache.kyuubi.{KyuubiException, KyuubiSQLException}
 import org.apache.kyuubi.config.KyuubiConf
 import org.apache.kyuubi.engine.{ApplicationOperation, KillResponse, ProcBuilder}
 import org.apache.kyuubi.engine.spark.SparkBatchProcessBuilder
+import org.apache.kyuubi.metrics.MetricsConstants.OPERATION_OPEN
+import org.apache.kyuubi.metrics.MetricsSystem
 import org.apache.kyuubi.operation.FetchOrientation.FetchOrientation
+import org.apache.kyuubi.operation.OperationState.OperationState
 import org.apache.kyuubi.operation.log.OperationLog
 import org.apache.kyuubi.session.KyuubiBatchSessionImpl
 import org.apache.kyuubi.util.ThriftUtils
 
+/**
+ * The state of batch operation is special. In general, the lifecycle of state is:
+ *
+ *                        /  ERROR
+ * PENDING  ->  RUNNING  ->  FINISHED
+ *                        \  CANCELED (CLOSED)
+ *
+ * We can not change FINISHED/ERROR/CANCELED to CLOSED, and it's different with other operation
+ * which final status is always CLOSED, so we do not use CLOSED state in this class.
+ * To compatible with kill application we combine the semantics of `cancel` and `close`, so if
+ * user close the batch session that means the final status is CANCELED.
+ */
```

Comment:
After some thought, the state lifecycle seems quite different between batch and sql. So I do this change in this pr. what do you think ? @turboFei @lightning-L @yaooqinn @pan3793 @yanghua @zhaomin1423

Reply:
I am thinking of `ERROR` or `FAILED`.